marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -238,45 +237,138 @@ private FailedRequest extractErrorFields(Response response) {
   }
 
   @Override
-  @Deprecated
-  public void connect(String host, int port, String database, String user, String password,Map<String,String> kerberosOptions,
-      Authentication authenType, SSLContext sslContext,
-      SSLHostnameVerifier verifier) {
-    connect(host, port, database, user, password, kerberosOptions, authenType, sslContext, null, verifier);
-  }
-
-  @Override
-  public void connect(String host, int port, String database, String user, String password,Map<String,String> kerberosOptions,
-      Authentication authenType, SSLContext sslContext, X509TrustManager trustManager,
-      SSLHostnameVerifier verifier) {
-    HostnameVerifier hostnameVerifier = null;
-    if (verifier == SSLHostnameVerifier.ANY) {
-      hostnameVerifier = new HostnameVerifier() {
-        @Override
-        public boolean verify(String hostname, SSLSession session) {
-          return true;
-        }
-      };
-    } else if (verifier == SSLHostnameVerifier.COMMON) {
-      hostnameVerifier = null;
-    } else if (verifier == SSLHostnameVerifier.STRICT) {
-      hostnameVerifier = null;
-    } else if (verifier != null) {
-      hostnameVerifier = new SSLHostnameVerifier.HostnameVerifierAdapter(verifier);
-    }// else {
-    //  throw new IllegalArgumentException(
-    //    "Null SSLContext but non-null SSLHostnameVerifier for client");
-    //}
-    connect(host, port, database, user, password, kerberosOptions, authenType, sslContext, trustManager, hostnameVerifier);
-  }
-
-  private void connect(String host, int port, String database, String user, String password, Map<String,String> kerberosOptions,
-                       Authentication authenType, SSLContext sslContext, X509TrustManager trustManager,
-                       HostnameVerifier verifier) {
-    logger.debug("Connecting to {} at {} as {}", new Object[]{host, port, user});
-
-    if (host == null) throw new IllegalArgumentException("No host provided");
-
+  public void connect(String host, int port, String database, SecurityContext securityContext){
+	String user = null;
+	Map<String, String> kerberosOptions = null;
+	String password = null;
+	Authentication type = null;
+	SSLContext sslContext = null;
+	SSLHostnameVerifier sslVerifier = null;
+	X509TrustManager trustManager = null;
+	String authorizationTokenValue = null;
```

Comment:
Can we remove some of these variables if we move the contents of the branches into methods?

Would the code be easier to read and thus understand if the indentation of the variables that remain aligned with line 249?